netunicorn-director/netunicorn-mediator/src/netunicorn/director/mediator/__main__.py

```diff
@@ -218,6 +253,180 @@ async def prepare_experiment_handler(
     return experiment_name
 
 
+@app.post("/api/v1/web/experiment/prepare", response_model=None)
```

Comment:
I think this function becomes too complicated for __main__, which is designed to just have handlers for web requests without significant logic
I think it's better to move the main long logic into engine if possible